azavea/franklin
# Include path fragment in open api spec and routes

## Overview

This PR makes it so that the API path fragment is respected everywhere. It accomplishes this through a helper method that takes an optional path prefix and the base path and builds a new endpoint input out of both.

### Checklist

- ~New tests have been added or existing tests have been modified~

### Testing Instructions

- Start the server with `/stac/api`, `stac/api`, and `stac/api/` `--api-path` arguments
- `http :9090/stac/api/collections` with each of them
- check `:9090/open-api/spec.yaml` for each of them
- start the server again with options like `--with-tiles` and `--with-transactions`
- check the spec again
- confirm that your path prefix is respected in all of the links

Closes #779  (if applicable)
